db/migrate/20180715202514_optimize_indices.rb

```diff
@@ -0,0 +1,45 @@
+class OptimizeIndices < ActiveRecord::Migration[5.1]
+  def change
+    remove_index :audits, name: "associated_index"
+    remove_index :audits, name: "auditable_parent_index"
+    remove_index :audits, name: "auditable_index"
+    remove_index :audits, [:user_id, :user_type]
+    add_index :audits, [:associated_type, :associated_id]
+    add_index :audits, [:user_type, :user_id]
+    add_index :audits, [:auditable_type, :auditable_id, :version]
+
+    remove_index :bookmarks, [:owner_id, :owner_type]
+    add_index :bookmarks, [:owner_type, :owner_id]
+
+    remove_index :fact_names, :ancestry
```

Comment:
`scope :root_only, -> { with_roots.where(:fact_names => {:ancestry => nil}) }`

Granted the "name" can follow, but there is just one use which does not use it.

Reply:
There is index `["ancestry", "name"]` that can be used for this as well